src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs

```diff
@@ -16,6 +16,155 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 
     public class ScriptSemanticsTests : CSharpTestBase
     {
+        private static CompilationReference TaskFacadeAssembly()
+        {
+            var corAssembly = @"
+namespace System.Runtime.CompilerServices {
+    public interface IAsyncStateMachine {
+        void MoveNext();
+        void SetStateMachine(IAsyncStateMachine iasm);
+    }
+    public struct AsyncTaskMethodBuilder {
+        public static AsyncTaskMethodBuilder Create() { return default(AsyncTaskMethodBuilder); }
+        public void SetStateMachine(IAsyncStateMachine stateMachine) { }
+    }
+    public struct AsyncTaskMethodBuilder<T> {
+        public static AsyncTaskMethodBuilder<T> Create() { return default(AsyncTaskMethodBuilder<T>); }
+        public void SetStateMachine(IAsyncStateMachine stateMachine) { }
+        public System.Threading.Tasks.Task<T> Task => null;
+        public void SetException(Exception e) { }
+        public void SetResult(T result) { }
+        public void AwaitOnCompleted<A, B>(ref A a, ref B b) { }
+        public void AwaitUnsafeOnCompleted<A, B>(ref A a, ref B b) { }
+        public void Start<A>(ref A a) { }
+    }
+}
+
+namespace System.Runtime.CompilerServices {
+    public class ExtensionAttribute {}
+}
+
+namespace System {
+    public delegate void Action();
+}
+namespace System.Runtime.CompilerServices {
+    public interface INotifyCompletion {
+        void OnCompleted(System.Action action);
+    }
+}
+
+namespace System.Threading.Tasks {
+    public class Awaiter<T>: System.Runtime.CompilerServices.INotifyCompletion {
+        public bool IsCompleted  => true;
+        public void OnCompleted(System.Action action) {}
+        public T GetResult() { throw new Exception(); }
+    }
+    public class Task<T> {}
+}
+namespace Hidden {
+public static class MyExtensions {
```

Comment:
We should also test scenario when MyExtensions class is in the global namespace, everything should work without any usings.